src/renderer/store/modules/playlists.js

```diff
@@ -1,32 +1,83 @@
 import { DBPlaylistHandlers } from '../../../datastores/handlers/index'
 
+function generateRandomPlaylistId() {
+  return `ft-playlist--${generateRandomUniqueId()}`
+}
+
+function generateRandomPlaylistName() {
+  return `Playlist ${new Date().toISOString()}-${Math.floor(Math.random() * 10000)}`
+}
+
+function generateRandomUniqueId() {
+  // To avoid importing `crypto` from NodeJS
+  return crypto.randomUUID ? crypto.randomUUID() : `id-${Date.now()}-${Math.floor(Math.random() * 10000)}`
+}
+
 const state = {
-  playlists: [
+  // Playlist loading takes time on app load (new windows)
+  // This is necessary to let components to know when to start data loading
+  // which depends on playlist data being ready
+  playlistsReady: false,
+  playlists: [],
+  defaultPlaylists: [
     {
       playlistName: 'Favorites',
-      protected: true,
-      videos: []
+      protected: false,
+      description: 'Your favorite videos',
+      videos: [],
+      _id: 'favorites',
     },
     {
-      playlistName: 'WatchLater',
-      protected: true,
-      removeOnWatched: true,
-      videos: []
-    }
-  ]
+      playlistName: 'Watch Later',
```

Comment:
Renaming 'Watch Later' to 'WatchLater' imported an empty playlist when importing an old version of FreeTube's playlists
![image](https://github.com/FreeTubeApp/FreeTube/assets/78101139/040c4455-65d2-42eb-99d2-8b0bf4a0e62f)

I also wonder if we should keep these default playlists protected. IMO it could be a good idea to keep Watch Later and Favorites as protected playlists, then we can also add translations to them

Reply:
You can reintroduce "default special playlists" later
Right now if you have protected, that means more functions & special playlist treatment need to be added and tested (which might still look incomplete due to playlist translation?)
Having an empty `WatchLater` (no space) playlist is an one time issue and I don't want to exclude that playlist (again that would be special playlist treatment)

Reply:
Also the idea of supporting multiple playlists is to have multiple "favorites" & "watch later" playlists (however they want to use their created playlist, as long term or temporary bookmark is their thing, we just need to enable them to do both). Until you can come up with features linked to "special playlists", those playlists shouldn't even be introduced, not to say protected.

Reply:
The main existent feature linked to a special playlist for Favorites is maintaining reverse compatibility regardless of user actions, no? That, and signalling to the user that those playlists do possess some special importance (reverse compatibility now, Watch Later / Favorites shortcut down the road). + Allows us to control the (translated) labels for these playlists at any point down the road for all users.

Reply:
Keeping non special playlists protected instead of allowing user to opt-in and choose the playlist(s) for later introduced features like "quick save video for purpose of long term bookmark and/or watch later"?

Reply:
I personally think customizing which playlist(s) the Watch Later button is for is a more debatable and lower priority feature than having one fixed & designated one. In any case, having one playlist that occupies the special position of representing legacy Favorites and one that occupies the special position of Watch Later (note: I'm fine with these two being the same) that are both protected is preferable. It provides a sense of constancy to the user and emphasizes their specialness for both reverse compatibility and Watch Later. The implementation details of the power user benefit of being able to delete and rename them is IMO not that important and can be revisited for when we do implement customizing which playlist(s) the Watch Later button is for.

Reply:
If anyone have developed/is developing the quick save video feature, I am fine it's not selectable at the beginning but the special playlist should be able to be hidden (when the feature is disabled).

Reverse compatibility can be done IMO via either by instructions in document (move all videos to one playlist) or by code (not sure how but that means another release required before this is merged).

Either way there is still no place for special playlists in this PR.